cpp/src_prims/metrics/scores.cuh

```diff
@@ -33,36 +33,55 @@
 #include <thrust/device_ptr.h>
 #include <thrust/reduce.h>
 
+#include <cuml/common/device_buffer.hpp>
```

Comment:
It would be much better to switch to using rmm objects (device_uvector), see this: https://github.com/rapidsai/raft/issues/97 and many of the algos have started that move